test/boost/querier_cache_test.cc

```diff
@@ -213,8 +213,10 @@ class test_querier_cache {
         const auto cache_key = make_cache_key(key);
 
         auto querier = make_querier<Querier>(range);
-        auto [dk, ck] = querier.consume_page(dummy_result_builder{}, row_limit, std::numeric_limits<uint32_t>::max(),
+        auto dk_ck = querier.consume_page(dummy_result_builder{}, row_limit, std::numeric_limits<uint32_t>::max(),
                 gc_clock::now(), db::no_timeout, query::max_result_size(std::numeric_limits<uint64_t>::max())).get0();
+        auto&& dk = dk_ck.first;
+        auto&& ck = dk_ck.second;
```

Comment:
Again I'm a bit confused about this idiom. Isn't it more natural to do something like
```
auto&& dc_ck = some_function();    // extend temporary lifetime
auto& dk = ...; // pointer to some part of that temporary
auto& ck = ...;
```

Reply:
It's the same thing. I prefer `auto&&` too in general.